src/App.tsx

```diff
@@ -1,10 +1,67 @@
+/* eslint-disable object-curly-newline */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { getTodos } from './api/todos';
+
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+
 import { UserWarning } from './UserWarning';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { TodoList } from './components/TodoList/TodoList';
+import { Notification } from './components/Notification/Notification';
 
-const USER_ID = 0;
+const USER_ID = 10587;
+// https://mate.academy/students-api/todos?userId=10587
 
 export const App: React.FC = () => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState<FilterType>(FilterType.All);
+  const [todos, setTodos] = useState<Todo[] | []>([]);
+
+  const handleCloseErrorMessage = useCallback(() => {
+    setErrorMessage('');
+  }, []);
+
+  const handleLoadTodos = useCallback(async () => {
+    try {
+      const todosFromApi = await getTodos(USER_ID);
+
+      setTodos(todosFromApi);
+    } catch {
+      setErrorMessage('Unable to load a todos');
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+    }
+  }, []);
+
+  useEffect(() => {
+    handleLoadTodos();
+  }, []);
+
+  const handleFilterSelected = () => {
+    setFilter(filter);
+  };
+
+  const completedTodos = todos.filter(todo => todo.completed);
+  const activeTodos = todos.filter(todo => !todo.completed);
```

Comment:
Let's use `useMemo` here
```suggestion
  const completedTodos = useMemo(() => todos.filter(todo => todo.completed, [todos]);
```

Reply:
done